### PR TITLE
Refactor request implementation

### PR DESCRIPTION
**Pull request by daytonn**
@kand @collin I need some eyes on this. The tests pass but I want to make sure I understand all the plumbing.

### CODE REVIEW

File: elements/campaign-display/campaign-display.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ class CampaignDisplay extends BulbsElement {
   render () {
     return (
       <CampaignDisplayRoot
-          data={this.state}
+          {...this.state}
```
Comment (kand) [inline elements/campaign-display/campaign-display.js]: This seems like unnecessary indirection, can we make this `campaign={this.state.campaign}` instead? Or is there a specific reason to keep it as a generic unpacking?

File: elements/campaign-display/components/campaign-display-image.js [hunk 2]
```diff
@@ -3,13 +3,14 @@ import React, { PropTypes } from 'react';
 export default function CampaignDisplayImage (props) {
   return (
     <div className='campaign-display-image'>
-      <a href={props.url}>
-        <img src={props.image} />
+      <a href={props.clickthrough_url}>
```
Comment (kand) [inline elements/campaign-display/components/campaign-display-image.js]: Something I noticed in avclub, we might need additional attributes on this anchor tag, like stuff for the analytics manager. Is there some way to we can pass arbitrary attributes to this anchor?

If not, or it's too sloppy, I think we can refactor out any issues the analytics manager has with resolving its attributes if those attributes don't live on the elements themselves.

Comment (collin) [inline elements/campaign-display/components/campaign-display-image.js]: which attributes?

So we couldn't get away with putting the attributes directly on the `<campagin-display>` as it is?

I think that would be ideal.

File: elements/campaign-display/campaign-display.js (hunk 1 above)
Comment (daytonn) [inline elements/campaign-display/campaign-display.js]: I suppose, it's really the same thing

File: elements/campaign-display/components/campaign-display-image.js (hunk 2 above)
Comment (daytonn) [inline elements/campaign-display/components/campaign-display-image.js]: I'd like to wait until there are actual requirements to inform this decision. I have no knowledge of any other attribute needs.

Comment (kand) [inline elements/campaign-display/components/campaign-display-image.js]: Actually, nevermind, if there's a problem with the analytics manager here, we should fix it in the analytics manager and keep this clean.

File: elements/campaign-display/components/campaign-display-root.js [hunk 3]
```diff
@@ -6,11 +6,11 @@ export default function CampaignDisplayRoot (props) {
   let component;
   switch (props.display) {
   case 'name':
-    component = <CampaignDisplayName name={props.data.name}/>;
+    component = <CampaignDisplayName name={props.campaign.name}/>;
     break;
 
   case 'image':
-    component = <CampaignDisplayImage name={props.data.image} />;
+    component = <CampaignDisplayImage {...props.campaign} />;
```
Comment (kand) [inline elements/campaign-display/components/campaign-display-root.js]: Same as above, can we explicitly name the attributes being used?

Comment (daytonn) [inline elements/campaign-display/components/campaign-display-root.js]: I suppose, but then I don't get to use the fancy new spread operator.

File: elements/campaign-display/campaign-display.js (hunk 1 above)
Comment (kand) [inline elements/campaign-display/campaign-display.js]: With the way it is now, I have to go to another file to figure out what would potentially be passed in here, and that's with prior knowledge of what the structure here is.

If we want to reduce cognitive load, we should be explicit about what goes into this component.

File: elements/campaign-display/components/campaign-display-root.js (hunk 3 above)
Comment (daytonn) [inline elements/campaign-display/components/campaign-display-root.js]: Actually, no. I don't think there's any value in manually unpacking the attributes. This is a common pattern for react and I don't see that it's an issue. This is what happens with a wrapper component which just delegates to more specific components.

**Comment by kand** [comment]
:+1:

**Pull request merged**